FirmataConstants.h

```diff
@@ -14,268 +14,84 @@
-#endif
-#define ENCODER_DATA            0x61 // reply with encoders current positions
-
-#ifdef SERVO_CONFIG
-#undef SERVO_CONFIG
-#endif
-#define SERVO_CONFIG            0x70 // set max angle, minPulse, maxPulse, freq
-
-#ifdef STRING_DATA
-#undef STRING_DATA
-#endif
-#define STRING_DATA             0x71 // a string message with 14-bits per char
-
-#ifdef STEPPER_DATA
-#undef STEPPER_DATA
-#endif
-#define STEPPER_DATA            0x72 // control a stepper motor
-
-#ifdef ONEWIRE_DATA
-#undef ONEWIRE_DATA
-#endif
-#define ONEWIRE_DATA            0x73 // send an OneWire read/write/reset/select/skip/search request
-
-#ifdef SHIFT_DATA
-#undef SHIFT_DATA
-#endif
-#define SHIFT_DATA              0x75 // a bitstream to/from a shift register
-
-#ifdef I2C_REQUEST
-#undef I2C_REQUEST
-#endif
-#define I2C_REQUEST             0x76 // send an I2C read/write request
-
-#ifdef I2C_REPLY
-#undef I2C_REPLY
-#endif
-#define I2C_REPLY               0x77 // a reply to an I2C read request
-
-#ifdef I2C_CONFIG
-#undef I2C_CONFIG
-#endif
-#define I2C_CONFIG              0x78 // config I2C settings such as delay times and power pins
-
-#ifdef REPORT_FIRMWARE
-#undef REPORT_FIRMWARE
-#endif
-#define REPORT_FIRMWARE         0x79 // report name and version of the firmware
-
-#ifdef EXTENDED_ANALOG
-#undef EXTENDED_ANALOG
-#endif
-#define EXTENDED_ANALOG         0x6F // analog write (PWM, Servo, etc) to any pin
-
-#ifdef PIN_STATE_QUERY
-#undef PIN_STATE_QUERY
-#endif
-#define PIN_STATE_QUERY         0x6D // ask for a pin's current mode and value
-
-#ifdef PIN_STATE_RESPONSE
-#undef PIN_STATE_RESPONSE
-#endif
-#define PIN_STATE_RESPONSE      0x6E // reply with pin's current mode and value
-
-#ifdef CAPABILITY_QUERY
-#undef CAPABILITY_QUERY
-#endif
-#define CAPABILITY_QUERY        0x6B // ask for supported modes and resolution of all pins
-
-#ifdef CAPABILITY_RESPONSE
-#undef CAPABILITY_RESPONSE
-#endif
-#define CAPABILITY_RESPONSE     0x6C // reply with supported modes and resolution
-
-#ifdef ANALOG_MAPPING_QUERY
-#undef ANALOG_MAPPING_QUERY
-#endif
-#define ANALOG_MAPPING_QUERY    0x69 // ask for mapping of analog to pin numbers
-
-#ifdef ANALOG_MAPPING_RESPONSE
-#undef ANALOG_MAPPING_RESPONSE
-#endif
-#define ANALOG_MAPPING_RESPONSE 0x6A // reply with mapping info
-
-#ifdef SAMPLING_INTERVAL
-#undef SAMPLING_INTERVAL
-#endif
-#define SAMPLING_INTERVAL       0x7A // set the poll rate of the main loop
-
-#ifdef SCHEDULER_DATA
-#undef SCHEDULER_DATA
-#endif
-#define SCHEDULER_DATA          0x7B // send a createtask/deletetask/addtotask/schedule/querytasks/querytask request to the scheduler
-
-#ifdef SYSEX_NON_REALTIME
-#undef SYSEX_NON_REALTIME
-#endif
-#define SYSEX_NON_REALTIME      0x7E // MIDI Reserved for non-realtime messages
-
-#ifdef SYSEX_REALTIME
-#undef SYSEX_REALTIME
-#endif
-#define SYSEX_REALTIME          0x7F // MIDI Reserved for realtime messages
+static const int SERIAL_MESSAGE =          0x60; // communicate with serial devices, including other boards
```

Comment:
This would also be a good time to align on terminology. I've been debating between _MESSAGE and _DATA and unifying. Traditionally it's been _DATA, but when I added Serial I thought _MESSAGE may make more sense since that's a midi term. I need to think about this some more.